File: Atividades/Palindromo/Palindromo.java
import java.util.Scanner;

class Palindromo{
    private String frase;

    public Palindromo(){
        this.frase = " ";
    }

    public Palindromo(String frase){
        this.frase = frase;
    }

    public void setFrase(String frase) {  this.frase = frase; }
    
    public void VerificarIterativo(){
        frase = frase.toLowerCase(); // usar tolower para deicar todas as letras minusculas
        int tamanho= frase.length(); // passar o tamanho da string
        for(int i =0; i< tamanho/2 ; i++ ){ //inicio do loop para verificar 
            if(frase.charAt(i) != frase.charAt(tamanho -1 - i) ){
                System.out.println("NAO");
                return;
            }
        }
    System.out.println("Sim");
    return;
    }
}

class TestDrivePalindromo{
    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        Palindromo palindromo = new Palindromo();
        String frase;
        do{
            frase = scanner.nextLine();
            if(frase.compareToIgnoreCase("FIM")==0 ){ break;  }
            palindromo.setFrase(frase);
            palindromo.VerificarIterativo();
        }while(true);
    }
}

File: Atividades/Palindromo/Palindromo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>


typedef struct{
    char frase[100];
    int tamanho;
}String;

/*void PalindromoIterativo(String *s){
    int i, j;
    for(i = 0, j = s->tamanho - 1; i < s->tamanho / 2; i++, j--){
        if(tolower(s->frase[i]) == tolower(s->frase[j])){
            printf("NAO\n");
            return;
        }
    }
    printf("SIM\n");
}*/

int PalindromoRecursivo(char *s, int tamanho, int index){
    if(index >= tamanho / 2){
        printf("SIM\n");
        return 1;
    }else if(tolower(s[index]) != tolower(s[tamanho - index - 1])){
        printf("NAO\n");
        return 0;
    }
    else{
        return palindromo(s, tamanho, index +    1);
    }
}
int main(){
    String s;
    
    do{
        fgets(s.frase, 100, stdin);
        s.tamanho = strlen(s.frase);
        if(strcmp(s.frase, "FIM") == 0){break;}
        //PalindromoInteretivo(&s);
        PalindromoRecursivo(s.frase, s.tamanho, 0);
    }while(1);


return 0;
}